### PR TITLE
Api fix

### PR DESCRIPTION
**Pull request by mubashir1osmani**
Incorrect request for beta o1 models from API key (unsupported max_tokens parameter)

1. added external parameter (max_completion_tokens) to replace max_tokens, for API users using the o1-mini model

Issue here: https://github.com/trypear/pearai-app/issues/128

----

> [!IMPORTANT]
> Fixes parameter issue for `o1-mini` model in `OpenAI.ts` by replacing `max_tokens` with `max_completion_tokens`.
> 
>   - **Behavior**:
>     - In `OpenAI.ts`, replace `max_tokens` with `max_completion_tokens` for `o1-mini` model in `_convertArgs()`.
>     - Deletes `max_tokens` if `o1-mini` model is used.
>   - **Misc**:
>     - Minor formatting change in `Fireworks.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=trypear%2Fpearai-submodule&utm_source=github&utm_medium=referral)<sup> for 254867a639ed217d67d5820d62e8bd7dd1e834b7. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: core/yarn.lock
Comment (nang-dev) [inline core/yarn.lock:1]: Oh, p sure we dont want yarn.lock changes here

**Comment by nang-dev** [comment]
Appreciate this a ton! Can we remove the yarn changes, and then we can merge :D

File: core/yarn.lock
Comment (mubashir1osmani) [reply core/yarn.lock:1]: changes made

**Comment by nang-dev** [comment]
@mubashir1osmani sorry, i see that the yarn.lock is still in the changes!

**Comment by mubashir1osmani** [comment]
@nang-dev made all changes, works fine now

**Comment by Fryingpannn** [comment]
Yoyo, why fireworks deleted?

**Comment by mubashir1osmani** [comment]
there were no significant changes to fireworks so i removed it

**Comment by mubashir1osmani** [comment]
im sorry abt that

**Pull request closed without merging**